### DISCUSSION TITLE
Dataset with multiple graphs to be used with LightGCN model

### DISCUSSION THREAD

**Discussion by SourabhRanade** [pyg-team/pytorch_geometric#6786] [Q&A]
I was trying to make use the lightGCN model that has already been provided under the **'torch_geometric.nn.models'** section. 
The way I understand it, the model can only take a dataset with one single graph as input. 
Is it possible to make use of this model with a dataset that contains multiple small graphs. Something similar to the way PPI dataset is ?

**Comment by rusty1s** [comment]
`LightGCN` learns shallow embeddings of nodes, so it is not possible to transfer the results to new graphs. Other than that, I think you could merge your graphs into a single one via `Batch.from_data_list(dataset)` and go from there.

**Reply by SourabhRanade** [reply]
@rusty1s thank you. Is there an example for LightGCN like a notebook (to play around a little with the model using an example?)

**Reply by rusty1s** [reply]
Not yet, this is something I always wanted to add :( If you are interested, we could provide an example together.

**Reply by SourabhRanade** [reply]
@rusty1s , sorry for the late reply. Yes, I would like to do so!!

**Reply by rusty1s** [reply]
That was not late at all. Do you have a plan on how to approach this? Shall we start with reproducing one of the experiments from LightGCN?

**Reply by SourabhRanade** [reply]
Yes, I think reproducing one of the experiments/benchmarks results mentioned in the paper would be good start! 
Luckily, the datasets mentioned in the paper have already been provided in the official repository : [Official repository](https://github.com/gusye1234/LightGCN-PyTorch.git).
Unless you think using the already exisiting datasets makes more sense!

**Reply by rusty1s** [reply]
So as a first step, we would need to add the datasets to PyG, I assume. This shouldn't be too hard. Do you wanna prepare a draft for this?

**Reply by SourabhRanade** [reply]
Yes, I will start with this!

**Reply by rusty1s** [reply]
Amazing, thank you :)

**Reply by SourabhRanade** [reply]
@rusty1s , exams are going on, I have gone through their code ... I'll revert soon with the draft. Sorry for the delay :(

**Reply by SourabhRanade** [reply]
@rusty1s I was trying to understand the code more in depth from the above mentioned github repository of LighGCN. 
Here I was following the program flow for the example dataset **lastfm**. 

Inside the `dataloader.py`, there is a method `getSparseGraph` that gets called from the `model.py` . 
Here, they generate the indices, data along with the final sparse graph at the end. I think we can make use of these indexes and data to generate a PyG compatible graph (by small modifications and returning from their code a little early). 

I wanted your thoughts on this. 

P.S: I am sorry if I have missed out on something obvious. I am new to GNNs and am learning as I explore more.

**Reply by rusty1s** [reply]
I am not familiar with the LightGCN code base either, but I think we can just provide the datasets similar to how they do it as a `torch_geometric.dataset`, and then combine model and dataset in a training script (similar to all other examples in PyG).
